homebridge/thermostat.ts

```diff
@@ -0,0 +1,189 @@
+          case 'heat':
+          case 'aux':
+            return Characteristic.TargetHeatingCoolingState.HEAT
+          case 'cool':
+            return Characteristic.TargetHeatingCoolingState.COOL
+        }
+      },
+      setValue: (targetHeatingCoolingState: number) => {
+        const mode: ThermostatMode | undefined = (():
+          | ThermostatMode
+          | undefined => {
+          switch (targetHeatingCoolingState) {
+            case Characteristic.TargetHeatingCoolingState.OFF:
+              return 'off'
+            case Characteristic.TargetHeatingCoolingState.HEAT:
+              return 'heat'
+            case Characteristic.TargetHeatingCoolingState.COOL:
+              return 'cool'
+            default:
+              return
+          }
+        })()
+        if (!mode) {
+          this.logger.error(
+            `Couldn’t match ${targetHeatingCoolingState} to a recognized mode string.`
+          )
+          return
+        }
+        this.logger.info(`Setting ${this.device.name} mode to ${mode}`)
+
+        return this.device.setInfo({ device: { v1: { mode } } })
+      },
+    })
+    // Only allow 'TargetHeatingCoolingState's which can be mapped to Ring modes
+    // Specifically, this omits .AUTO
+    this.getService(Service.Thermostat)
+      .getCharacteristic(Characteristic.TargetHeatingCoolingState)
+      .setProps({
+        validValues: [
+          Characteristic.TargetHeatingCoolingState.OFF,
+          Characteristic.TargetHeatingCoolingState.HEAT,
+          Characteristic.TargetHeatingCoolingState.COOL,
+        ],
+      })
+
+    this.registerCharacteristic({
+      characteristicType: Characteristic.CurrentTemperature,
+      serviceType: Service.Thermostat,
+      getValue: async () => {
+        const temperature = await this.getTemperatureFromComponentDevices()
+        if (!temperature) {
+          return
+        }
+        // Documentation: https://developers.homebridge.io/#/characteristic/CurrentTemperature
+        // 'Characteristic.CurrentTemperature' supports 0.1 increments
+        return Math.round(temperature * 10) / 10
+      },
+    })
+
+    this.registerCharacteristic({
+      characteristicType: Characteristic.TargetTemperature,
+      serviceType: Service.Thermostat,
+      getValue: ({ setPoint }) => {
+        return setPoint
+      },
+      setValue: (setPoint: number) => {
+        this.logger.info(
+          `Setting ${this.device.name} target temperature to ${setPoint}`
+        )
+
+        // Documentation: https://developers.homebridge.io/#/characteristic/TargetTemperature
+        // 'Characteristic.TargetTemperature' has a valid range from 10 to 38 degrees celsius,
+        // but devices may support a different range. When limits differ, accept the more strict.
+        const setPointMin = Math.max(this.device.data.setPointMin || 10, 10),
+          setPointMax = Math.min(this.device.data.setPointMax || 38, 38)
```

Comment:
Thoughts on overriding the minValue and maxValue on the `TargetTemperature` if they are present on the device data?

Reply:
👍 Good idea! Added in https://github.com/dgreif/ring/pull/331/commits/ce3072ac677d7aee2e0529c278476262204a4bb6